### PR TITLE
Add LiteX PWM driver

### PR DESCRIPTION
**Pull request by mateusz-holenko**
This adds LiteX PWM driver and enables it in litex_vexriscv board.

### CODE REVIEW

**Comment by mateusz-holenko** [comment]
Rebased to resolve conflicts after merging #18649.

File: drivers/pwm/pwm_litex.c [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+}
+
+static const struct pwm_driver_api pwm_litex_driver_api = {
+	.pin_set             = pwm_litex_pin_set,
+	.get_cycles_per_sec  = pwm_litex_get_cycles_per_sec,
+};
+
+/* Device Instantiation */
+
+/* LiteX regisers use only first bit from 4-bit register, that's why they
```
Comment (kgugala) [inline drivers/pwm/pwm_litex.c]: looking at the code, I think you meant bytes not bits

**Review by kgugala — CHANGES_REQUESTED** [review]
small comment issue, otherwise LGTM

File: drivers/pwm/pwm_litex.c (hunk 1 above)
Comment (mateusz-holenko) [reply drivers/pwm/pwm_litex.c]: You're right. It's fixed now.

**Review by kgugala — APPROVED** [review]
LGTM

**Comment by jhedberg** [comment]
@mateusz-holenko there are some conflicts that need to be resolved before merging. Please rebase against latest master, thanks.

**Comment by mateusz-holenko** [comment]
@jhedberg I resolved the conflicts - all checks are now green and happy.

**Pull request merged**